Introduce a new WiFi manager (#15022)
zephyrproject-rtos/zephyr: samples/wifi/connect/README.rst

.. _wifi_sta_connect:

Wi-Fi: Auto Connect
####################

Overview
********

> Application demonstrating usage of STA connect APIs of Wi-Fi manager.

Reviewer: Use a complete sentence, and fix some reST formatting (need a blank line before a list):
```
This sample demonstrates using the Wi-Fi manager STA connect APIs
to connect a board to an access point (AP), following these steps:

1. Get STA config to obtain the specified AP
2. Open STA
3. Scan the specified AP
4. Connect to the specified AP
```

To make the board connect to an AP, the necessary steps are as follows.
1. Get STA config to obtain the specified AP
2. Open STA
3. Scan the specified AP
4. Connect to the specified AP

Requirements
************

* An access point with IEEE 802.11 a/b/g/n/ac support

Building and Running
********************

This sample can be found under :zephyr_file:`samples/wifi/connect` in
the Zephyr tree.

> See :ref:`wifi samples section <wifi-samples>` for details.

Reviewer: There are no build or run details in this reference document?

> See :ref:`wifi samples section <wifi-samples>` for details.

Reviewer: change to ``Wi-Fi samples section``

.. _wifi_sta_connect:

Wi-Fi: Auto Connect
####################

Overview
********

This sample application shows how to use the Wi-Fi manager APIs to connect an AP.
It demonstrates several main features: getting STA config, opening STA,
scanning, connecting an AP.

Requirements
************

* A board with Wi-Fi and flash support
* An access point with IEEE 802.11 a/b/g/n/ac support

Building and Running
********************

This sample can be found under :zephyr_file:`samples/wifi/connect` in
the Zephyr tree.

This sample can be built for the boards which support Wi-Fi manager,
in this example we will build it for the ``96b_ivy5661`` board:

.. zephyr-app-commands::
   :zephyr-app: samples/wifi/connect
   :board: 96b_ivy5661
   :goals: build flash
   :compact:

To run this sample, the STA must be configured in advance.

> To specified an open AP, please follow the command below.

Reviewer: ```suggestion
To specify an open AP, use the following command:
```

.. code-block:: console

    wifimgr set_config sta -n "MERCURY_3BC4" -p "" -a 10000

> To specified an WPA/WPA2-PSK secured AP, please follow the command below.

Reviewer: ```suggestion
To specify a WPA/WPA2-PSK secured AP, use the following command:
```
